src/Plugin/Action/BlockIP.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\rules\Plugin\Action\BlockIP.
+ */
+
+namespace Drupal\rules\Plugin\Action;
+
+use Drupal\ban\BanIpManagerInterface;
+use Drupal\Component\Utility\String;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\rules\Engine\RulesActionBase;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Provides the 'Block IP' action.
+ *
+ * @Action(
+ *   id = "rules_block_ip",
+ *   label = @Translation("Block IP"),
+ *   category = @Translation("System"),
+ *   context = {
+ *     "ip" = @ContextDefinition("string",
+ *       label = @Translation("IP Address")
+ *     )
```

Comment:
In D7 the IP parameter was optional and if not provided the IP address of the current user would be used. Ithink we should do the same here again.